packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,465 @@
+            onFocus: (e: React.SyntheticEvent<HTMLElement>) => {
+              if (item.props.level === 1) return;
+              data.handleOnFocus(e);
+            },
+          })
+        : null;
+    }
+
+    return null;
+  }
+}
+
+const InnerElementType = ({ children, style }) => {
+  const context = React.useContext(InnerElementContext);
+  const { height, stickyItemSize, stickyTopIds, stickyBottomIds, onClickSticky } = context;
+
+  if (!children.length) return <></>;
+
+  const data = children[0]?.props?.data;
+  const { renderedItems } = data;
+
+  const stickyTopItemsIndex = stickyTopIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+  const stickyBottomItemsIndex = stickyBottomIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+
+  return (
+    <div
+      style={{
+        ...style,
+        height: Math.max(height, style.height),
+      }}
+    >
+      {stickyTopItemsIndex.map((index: number) => {
+        const item = renderedItems[index];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: 2,
+            top: index * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+      {children}
+      {stickyBottomItemsIndex.map((itemIndex: number, index: number, arr: number[]) => {
+        const item = renderedItems[itemIndex];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: 111,
+            top: height - (arr.length - index) * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+    </div>
+  );
+};
+
+const calcRenderedHeight = (renderedItems: React.ReactElement[], stickyItemSize: number, itemSize: number) => {
+  let renderedHeight = 0;
+  renderedItems.forEach(item => {
+    renderedHeight = renderedHeight + (item.props.level === 1 ? stickyItemSize : itemSize);
+  });
+  return renderedHeight;
+};
+
+interface ReRenderStickyTreeProps {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  stickyItemSize: number;
+  itemSize: number;
+  height: number;
+  toplvlItemsId: string[];
+  onActiveItemsIdChange: (idsToRemove?: string[], idsToAdd?: string[]) => void;
+}
+
+interface ItemDataType {
+  renderedItems: React.ReactElement<TreeItemProps>[];
+  isNonSticky: (item: React.ReactElement<TreeItemProps>) => boolean;
+  onClick: (event: React.SyntheticEvent<HTMLElement>, itemProps: TreeItemProps) => void;
+  handleOnFocus: (e: React.SyntheticEvent<HTMLElement>) => void;
```

Comment:
nit: might as well be consistent
```suggestion
  handleOnFocus: (event: React.SyntheticEvent<HTMLElement>) => void;
```